packages/fuels-core/src/lib.rs

```diff
@@ -203,27 +201,27 @@ impl Default for Token {
 
 pub trait Tokenizable {
     /// Converts a `Token` into expected type.
-    fn from_token(token: Token) -> Result<Self, InstantiationError>
+    fn from_token(token: Token) -> Result<Self, Error>
     where
         Self: Sized;
     /// Converts a specified type back into token.
     fn into_token(self) -> Token;
 }
 
-pub fn try_from_bytes<T>(bytes: &[u8]) -> Result<T, crate::errors::Error>
+pub fn try_from_bytes<T>(bytes: &[u8]) -> Result<T, Error>
 where
     T: Parameterize + Tokenizable,
 {
     let token = ABIDecoder::decode_single(&T::param_type(), bytes)?;
 
-    // Not inlined so that the error conversion InstantiationError -> Error
+    // Not inlined so that the error conversion Error -> Error
```

Comment:
Since InstantiationError is now part of the 'Error', I think this can now be inlined? i.e.

```rust
let token = ABIDecoder::decode_single(&T::param_type(), bytes)?;

T::from_token(token)
```